hedera-node/src/main/java/com/hedera/services/store/contracts/WorldLedgers.java

```diff
@@ -196,7 +197,13 @@ public Address getAddressOrAlias(final Address address) {
 			alias = staticEntityAccess.alias(sourceId);
 		}
 		if (!alias.isEmpty()) {
-			return Address.wrap(Bytes.wrap(alias.toByteArray()));
+			if (alias.size() == 20) {
+				return Address.wrap(Bytes.wrap(alias.toByteArray()));
+			} else if (alias.size() == 35 && alias.startsWith(ByteString.copyFrom(new byte[] {0x3a, 0x21}))) {
+				return Address.wrap(Bytes.wrap(AliasManager.calculateEthAddress(alias.substring(2).toByteArray()).toByteArray()));
```

Comment:
This will now be 

```
return Address.wrap(Bytes.wrap(EthTxSigs.recoverAddressFromPubKey(alias.substring(2).toByteArray()))`
```
post-https://github.com/hashgraph/hedera-services/pull/3258, right?

Reply:
Yes.